inception-project/inception
# [WIP] 785 log external search

**What's in the PR**
* This PR adds logging of external searches made by users. Closes #785 .

**How to test manually**
1. Create and activate a project configured to use external search.
2. From the main Inception page, click “Search”.
3. Type a query into the search box, say 'christmas`, and hit “Search”.
4. Confirm that there is a corresponding console log for `ExternalSearchQueryEvent`  that looks something like:
`2018-12-14 11:26:15 INFO [admin] EventRepositoryImpl - LoggedEvent [created=Fri Dec 14 11:26:15 PST 2018, event=ExternalSearchQueryEvent, user=admin, project=1, annotator=admin, details={"query":"christmas"}]`
Confirm that all the fields are correct.
5. Open the database to check in the `Inception` database, table `logged_event` to make sure the search is logged. Below are commands for MySQL.
5.1 `mysql --user=root --password` to log into database as root.
5.2 `use inception` to switch to database for inception.
5.3 `SELECT * FROM logged_event ORDER BY created ASEC LIMIT 1` to see the last created log. Confirm that it is an ExternalSearchQueryEvent for `event` column and that `user`, `project` and `details` (query string) are correct.

**Automatic testing**
* [ ] PR includes unit tests (not immediately obvious how to unit test it; current search logging doesn't have tests as a model)

**Documentation**
* [ ] PR updates documentation (this probably doesn't need to be documented, unless we have a list somewhere of all logged events)
